frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyCodeHandler.java

```diff
@@ -254,8 +239,33 @@ private void processSuccessfulCodeRequest(
         }
         codeStorageService.deleteOtpCode(session.getEmailAddress(), notificationType);
 
+        var metadataPairArray =
+                metadataPairs(notificationType, journeyType, codeRequest, loginFailureCount, true);
         auditService.submitAuditEvent(
-                FrontendAuditableEvent.CODE_VERIFIED, auditContext, metadataPairs);
+                FrontendAuditableEvent.CODE_VERIFIED, auditContext, metadataPairArray);
+    }
+
+    private AuditService.MetadataPair[] metadataPairs(
+            NotificationType notificationType,
+            JourneyType journeyType,
+            VerifyCodeRequest codeRequest,
+            Integer loginFailureCount,
+            Boolean successfulRequest) {
+        AuditService.MetadataPair[] basicMetadataPairs = {
+            pair("notification-type", notificationType.name()),
+            pair("account-recovery", journeyType.equals(JourneyType.ACCOUNT_RECOVERY)),
+            pair("journey-type", String.valueOf(journeyType))
+        };
+        List<AuditService.MetadataPair> elementsList = new ArrayList<>(List.of(basicMetadataPairs));
+        if (notificationType.equals(MFA_SMS)) {
+            elementsList.add(pair("mfa-type", MFAMethodType.SMS.getValue()));
+            elementsList.add(pair("loginFailureCount", loginFailureCount));
+            elementsList.add(pair("MFACodeEntered", codeRequest.code()));
+            if (Boolean.FALSE.equals(successfulRequest)) {
```

Comment:
Is there a reason we're using boxed booleans in this method rather than primitive booleans? We could write `if (!successfulRequest) {...}`

Reply:
Yeah because sonarcloud complains about that first syntax 😢